tests/test_userprofile.py

```diff
@@ -132,3 +134,30 @@ def test_order_with_lines_pagination(admin_client, order_list):
     url = reverse('profile:details')
     response = admin_client.get(url, data)
     assert response.status_code == 200
+
+
+def test_format_address(billing_address):
+    formatted_address = format_address(billing_address)
+    address_html = '<br>'.join(map(str, formatted_address['address_lines']))
+    context = Context({'address': billing_address})
+    tpl = Template(
+        '{% load i18n_address_tags %}'
+        '{% format_address address %}')
```

Comment:
Can't we just call the template tag as a function?

Reply:
Intention here is to assert that template follows contract, eg. address contents are separated by either `<br>`'s or commas, and if inline template contains `inline-address` css class.  Admittedly this ties us to template implementation that will fail when `{{ address_lines|join:'<br>' }}` or `inline_address` gets replaced with something else, but doing so will also mess with dashboard as we are now reusing {% format_address %} there, so something to grab attention that editing this template should be done if you know what you are doing is good imho.

Reply:
How would this be different when calling `format_address(address)`? Currently you're also testing that Django's templating engine works.

Reply:
This is because not all formatting happens in tag's logic. Depending on inline arg being true or false template renders lists of elements differently. If we've tested just returned value, we would know if tag passes `inline` to template and elements of `address` were formatted via `i18naddress.format_address`, but that leaves out the `Part<br>Other Part` vs `Parth, Other Part` difference, because that only happens in template render..